src/combinatorics/simple_graph/basic.lean

```diff
@@ -317,6 +317,30 @@ instance decidable_mem_edge_set [decidable_rel G.adj] :
 instance edges_fintype [decidable_eq V] [fintype V] [decidable_rel G.adj] :
   fintype G.edge_set := subtype.fintype _
 
+section from_edge_set
+
+variable (s : set (sym2 V))
+
+/--
+`from_edge_set` constructs a `simple_graph` from a set of edges, without loops.
+-/
+def from_edge_set : simple_graph V :=
+{ adj := ne ⊓ sym2.to_rel s,
+  symm := λ v w h, ⟨h.1.symm, sym2.to_rel_symmetric s h.2⟩}
+
+@[simp] lemma from_edge_set_adj : (from_edge_set s).adj v w ↔ v ≠ w ∧ ⟦(v, w)⟧ ∈ s := iff.rfl
+
```

Comment:
Thanks! Do you think this should be `@[simp]` or not? To me, the right side definitely feels "simpler", but you didn't include the tag.

Reply:
I guess there's a confluence issue here. If you simp the expression `⟦(v, w)⟧ ∈ (from_edge_set s).edge_set`, then through `from_edge_set_adj` and other simp lemmas you get `v ≠ w ∧ ⟦(v, w)⟧ ∈ s`, but if you go through `edge_set_from_edge_set` you get `⟦(v, w)⟧ ∈ s ∧ v ≠ w`.

It makes sense that they should both be simp lemmas, and maybe the best solution I see is to change the definition of `from_edge_set` to use `sym2.to_rel s ⊓ ne`. Another option is to alter `from_edge_set_adj` to have the `and_comm` of the conclusion.

Reply:
I hope you don't mind, but I went ahead did the mechanical changes to ensure the simp set is confluent. There are a few people looking forward to this PR being merged, otherwise I would have let you decide whether you wanted to do it (I apologize for my manners here!).

Once I do a final check, I go and merge it.

Reply:
By the way, this is what I meant about simp confluence. Since removing either simp lemma allows simp to simplify, we want both of the following simp invocations to yield the same normal form:
```lean
example (s : set (sym2 V)) (u v : V) : ⟦(u, v)⟧ ∈ (from_edge_set s).edge_set :=
begin
  --simp [-from_edge_set_adj],
  --simp [-edge_set_from_edge_set],
end
```

Reply:
You're right, I haven't thought that through. Thank you!